src/sentry/issues/issue_occurrence.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence
+
+from dateutil.parser import parse as parse_date
+
+from sentry import nodestore
+from sentry.types.issues import GroupType
+from sentry.utils.dates import ensure_aware
+
+
+@dataclass(frozen=True)
+class IssueEvidence:
+    name: str
+    value: str
+    important: bool
+
+    def to_dict(
+        self,
+    ) -> Mapping[str, Any]:
+        return {
+            "name": self.name,
+            "value": self.value,
+            "important": self.important,
+        }
+
+
+@dataclass(frozen=True)
+class IssueOccurrence:
+    """
+    A class representing a specific occurrence of an issue. Separate to an `Event`. An `Event` may
+    have 0-M `IssueOccurrences` associated with it, and each `IssueOccurrence` is associated with
+    one `Event`.
+
+    Longer term, we might change this relationship so that each `IssueOccurrence` is the primary
+    piece of data that is passed around. It would have an `Event` associated with it.
+    """
+
+    id: str
+    # Event id pointing to an event in nodestore
+    event_id: str
+    fingerprint: Sequence[str]
+    issue_title: str
+    # Exact format not decided yet, but this will be a string regardless
+    subtitle: str
+    resource_id: str | None
+    # Extra context around how the problem was detected. Used to display grouping information on
+    # the issue details page, and will be available for use in UI customizations.
+    evidence_data: Mapping[str, Any]
+    # Extra context around the problem that will be displayed as a default in the UI and alerts.
+    # This should be human-readable. One of these entries should be marked as `important` for use
+    # in more space restricted integrations.
+    evidence_display: Sequence[IssueEvidence]
+    type: GroupType
+    detection_time: datetime
+
+    def __post_init__(self) -> None:
+        object.__setattr__(self, "detection_time", ensure_aware(self.detection_time))
```

Comment:
This is kind of a gross way to enforce that the datetime has a timezone. Wondering if it would be better to just check whether it's tz aware and throw a `ValueError` if not, rather than silently coercing.